lib/idris-ide-mode.coffee

```diff
@@ -9,15 +9,16 @@ class IdrisIdeMode extends EventEmitter
   buffer: ''
   idrisBuffers: 0
 
-  constructor: ->
-    pathToIdris = atom.config.get("language-idris.pathToIdris")
-    @process = spawn pathToIdris, ['--ide-mode']
-    @process.on 'error', @error
-    @process.on 'exit', @exited
-    @process.on 'close', @exited
-    @process.on 'disconnect', @exited
+  start: ->
+    if (not @process?) || @process.killed
+      pathToIdris = atom.config.get("language-idris.pathToIdris")
+      @process = spawn pathToIdris, ['--ide-mode']
+      @process.on 'error', @error
+      @process.on 'exit', @exited
+      @process.on 'close', @exited
+      @process.on 'disconnect', @exited
```

Comment:
Looks like, in place of the `connected` property that wasn't working for you, you're watching for the process to die via all the various events?